### PR TITLE
Feat: add support for json, string and binary message keys

### LINKED ISSUE #95: Feature Request: Support Returning Kafka Message Keys
**Is your feature request related to a problem? Please describe.**
Enhance the datasource so that if a message includes a non-null key, it is returned alongside the value.

- Add a key column in the resulting dataframe.
- If the key is null, the column can be empty for that row.
- Keys should be returned as strings (if binary, base64 encode them).
- Optionally, allow users to toggle “Include Keys” in the query editor for cleaner dataframes if keys are not needed.

**Describe the solution you'd like**
- Pre-processing Kafka topics to copy keys into message values — adds operational complexity and defeats the purpose of Kafka keys.
- Ignoring keys — not acceptable for teams that rely on them for partitioning or identity.

**Describe alternatives you've considered**

**Which sides should be considered to develop?**

- [x] Frontend(React)
- [x] Backend(Golang)

**Additional context**
- Keys are often used for correlation IDs, partitioning strategies, or routing hints — visualizing them makes troubleshooting much easier.
- Example dataframe with key:

| key        | value.host | value.metric | value.cpu |
|------------|------------|--------------|-----------|
| tenant-001 | srv-01     | load         | 0.75      |
| tenant-002 | srv-02     | load         | 0.81      |

### PR DESCRIPTION
**Pull request by hoptical**
Closes #95 
This pull request introduces comprehensive support for Kafka message keys in the Grafana Kafka datasource, allowing users to choose how message keys are processed and displayed. The new functionality includes support for ignoring keys, treating them as strings, or parsing and flattening them as JSON objects. The implementation includes backend changes, frontend UI updates, documentation, and thorough unit testing to ensure robust behavior and backward compatibility.

**Kafka message key support and configuration:**

* Added support for message key processing modes: None (ignore), String (single column), and JSON (flattened object with "key." prefix). This is configurable via the UI, API, and CLI, with the default being None for backward compatibility. (`README.md`, `docs/MESSAGE_KEYS.md`, `pkg/plugin/plugin.go`, `pkg/plugin/stream_manager.go`, `src/types.ts`, `src/QueryEditor.tsx`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R46) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R100-R103) [[3]](diffhunk://#diff-0599a48881e856c6ddfd4a2efafd6c80a699413ad3822e754bc04d6c93b2252aR1-R252) [[4]](diffhunk://#diff-fc137d62147eb6a9c89c0566479ff219320ca691f2505631ebacbe513bcc8108R217-R218) [[5]](diffhunk://#diff-fc137d62147eb6a9c89c0566479ff219320ca691f2505631ebacbe513bcc8108R666) [[6]](diffhunk://#diff-fc137d62147eb6a9c89c0566479ff219320ca691f2505631ebacbe513bcc8108R687-R689) [[7]](diffhunk://#diff-150943e3bd5791e47e6fc4a10f9f77841c8b437b6cd5f6c03508b697a8a9af08R97)

* Updated the example Go producer to allow generating messages with string, JSON, or no keys, including validation and message construction logic. (`example/go/producer.go`) [[1]](diffhunk://#diff-49eae8a10e9a360ccbf86bc0aca0bf9dbefe9faa1d73f6fc0276e9e7e4dac758R82) [[2]](diffhunk://#diff-49eae8a10e9a360ccbf86bc0aca0bf9dbefe9faa1d73f6fc0276e9e7e4dac758R92-R97) [[3]](diffhunk://#diff-49eae8a10e9a360ccbf86bc0aca0bf9dbefe9faa1d73f6fc0276e9e7e4dac758R279-R302) [[4]](diffhunk://#diff-49eae8a10e9a360ccbf86bc0aca0bf9dbefe9faa1d73f6fc0276e9e7e4dac758L285-R312)

**Backend implementation and data flow:**

* Extended the `KafkaMessage` struct to include decoded and raw key fields, and updated the consumer logic to capture raw keys for later decoding. (`pkg/kafka_client/client.go`) [[1]](diffhunk://#diff-8b8230fd8a6aa295e22d0a215a4cd273af9d04daa905140b9cd00542200da5baR103-R104) [[2]](diffhunk://#diff-8b8230fd8a6aa295e22d0a215a4cd273af9d04daa905140b9cd00542200da5baR440-R446)

**Testing and validation:**

* Added comprehensive unit tests for key decoding (string, JSON, none, and edge cases) and for JSON flattening with key prefixes, ensuring correctness and robustness. (`pkg/plugin/key_utils_test.go`, `pkg/plugin/json_utils_test.go`) [[1]](diffhunk://#diff-3b81e86a3a6e0dbd4ec637377240777cc595a44ca45870620760f37efd8b5fc7R1-R265) [[2]](diffhunk://#diff-16d4ef87687ed7e59a84269548e84fc24d0d043a0350129be2146a9ab95b89faR255-R337)

**Documentation:**

* Added detailed documentation (`docs/MESSAGE_KEYS.md`) describing the feature, configuration options, backend/frontend architecture, error handling, field ordering, troubleshooting, and example usage.

These changes provide users with fine-grained control over how Kafka message keys are handled and visualized, improving the flexibility and utility of the datasource while maintaining backward compatibility.

## Summary by CodeRabbit

* **New Features**
  * Kafka message key support with a Key Format option (None, String, JSON, Base64); Key Format selector in the query editor and key-derived columns added to streams when enabled.

* **Documentation**
  * New MESSAGE_KEYS guide plus README updates with usage, examples, and troubleshooting; example producer docs updated.

* **Tests**
  * Expanded unit and end-to-end tests covering key decoding, JSON flattening, Base64 handling, and UI/streaming behavior.

* **Chores**
  * Added markdown linting, CI lint step, and pre-commit config.

### CODE REVIEW

File: pkg/plugin/stream_manager.go [hunk 1]
```diff
@@ -778,6 +890,9 @@ func (sm *StreamManager) ProcessMessage(
 	offsetField.Set(0, msg.Offset)
 	fields = append(fields, offsetField)
 
+	//Decode and process message key if configured
```
[the remark this answers is not in the document]
Comment (hoptical) [reply pkg/plugin/stream_manager.go]: fixed.
